js/components/timer.js

```diff
@@ -0,0 +1,194 @@
+      this.hours = Math.floor(totalMinutes / 60);
+      this.minutes = totalMinutes % 60;
+    },
+    toSeconds: function(hours, minutes, seconds) {
+      return (+hours * 60 * 60) + (+minutes * 60) + +seconds;
+    },
+    start: function() {
+      if (this.timerStatus === 'running') {
+        return;
+      }
+
+      this.timerStatus = 'running';
+      this.startTimestamp = moment().unix();
+      Fliplet.App.Storage.set(this.name, this.startTimestamp);
+      this.setInterval();
+    },
+    stop: function() {
+      if (this.timerStatus === 'stopped') {
+        return;
+      }
+
+      this.value = this.updateValue();
+      Fliplet.App.Storage.remove(this.name);
+      this.timerStatus = 'stopped';
+      this.stopInterval();
+    },
+    reset: function() {
+      this.value = 0;
+
+      if (this.timerStatus === 'running') {
+        Fliplet.App.Storage.remove(this.name);
+        this.timerStatus = 'stopped';
+        this.stopInterval();
+      }
+
+      this.setDisplayValues(this.initialValue);
+    },
+    setInterval: function() {
+      var $vm = this;
+
+      if (this.type === 'stopwatch') {
+        this.timerInterval = setInterval(function() {
+          var totalSeconds = $vm.updateValue();
+
+          $vm.setDisplayValues(totalSeconds);
+        }, 500);
+      } else if (this.type === 'timer') {
+        this.timerInterval = setInterval(function() {
+          if (!$vm.hours && !$vm.minutes && !$vm.seconds) {
+            Fliplet.UI.Toast({
+              type: 'minimal',
+              message: 'Countdown Timer has reached 0',
+              actions: [{
+                label: 'Dismiss',
+                action: () => {
+                  Fliplet.UI.Toast.dismiss();
+                  $vm.reset();
+                }
+              }],
+              duration: false, // Ensures the toast message doesn't auto-dismiss
+              tapToDismiss: false // Ensures the toast message is only dismissed through the action button
+            });
+
+            $vm.stop();
+          } else {
+            var totalSeconds = $vm.initialValue - $vm.updateValue();
+
+            $vm.setDisplayValues(totalSeconds);
+          }
+        }, 500);
+      }
+    },
+    stopInterval: function() {
+      clearInterval(this.timerInterval);
+      this.timerInterval = null;
+    },
+    get: function() {
+      return this.updateValue();
+    },
+    set: function(data) {
+      if (!/^(\d+.)*(\d+)$/.test(data)) {
+        return;
+      }
+
+      data = parseFloat(data).toFixed(3);
+
+      this.value = this.type === 'timer' ? Math.min(this.initialValue, data) : Math.max(data, 0);
+      this.setDisplayValues(this.value);
+
+      if (this.timerStatus === 'running') {
+        this.startTimestamp = moment().unix();
+        Fliplet.App.Storage.set(this.name, this.startTimestamp);
+      }
+    },
+    updateValue() {
+      return Math.floor(this.value) + (moment().unix() - this.startTimestamp);
```

Comment:
I don't think we need to round this. If you do, let me know why.

```suggestion
      return this.value + (moment().unix() - this.startTimestamp);
```